src/main/java/core/basesyntax/serviceimpl/ShopServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction: transactions) {
+            FruitTransaction.Operation operation = transaction.getOperation();
+            OperationHandler handler = operationStrategy.getHandler(operation);
+            handler.apply(transaction);
+        }
+    }
```

Comment:
Consider adding exception handling within the `process` method. If `operationStrategy.getHandler(operation)` returns null or if an exception occurs in `handler.apply(transaction)`, the program will crash. It's recommended to add a try-catch block and throw a custom exception with an informative message if an error occurs.